File: Excel/700-799/737/737 Challenge.py
import pandas as pd
import re

path = "700-799/737/737 Split Text on Transition.xlsx"
input = pd.read_excel(path, usecols="A", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="B:F", skiprows=1, nrows=5)

value_cols = input.iloc[:, 0].astype(str).str.split(
    r'(?<=[A-Za-z])-(?=\d)|(?<=\d)-(?=[A-Za-z])',
    expand=True
)
result = value_cols.set_axis(
    [f'Value{i+1}' for i in range(value_cols.shape[1])],
    axis=1
)

print(result.equals(test)) # True

File: Excel/700-799/737/737 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/737/737 Split Text on Transition.xlsx"
input = read_excel(path, range = "A2:A6")
test = read_excel(path, range = "B2:F6")

result = input %>%
  mutate(
    Value = str_split(Text, "(?<=[A-Za-z])-(?=\\d)|(?<=\\d)-(?=[A-Za-z])")
  ) %>%
  unnest_wider(Value, names_sep = "") %>%
  select(-Text)

all.equal(result, test)
# [1] TRUE
